kotlinx-coroutines-core/jvm/src/debug/AgentPremain.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.debug
+
+import kotlinx.coroutines.debug.internal.DebugProbesImpl
+import sun.misc.*
+import java.lang.instrument.*
+import java.lang.instrument.ClassFileTransformer
+import java.security.*
+
+@Suppress("unused")
+internal object AgentPremain {
+
+    public var isInstalledStatically = false
+
+    private val enableCreationStackTraces =
+        System.getProperty("kotlinx.coroutines.debug.enable.creation.stack.trace")?.toBoolean()
+            ?: DebugProbesImpl.enableCreationStackTraces
+
+    @JvmStatic
+    public fun premain(args: String?, instrumentation: Instrumentation) {
+        isInstalledStatically = true
+        instrumentation.addTransformer(DebugProbesTransformer)
+        DebugProbesImpl.enableCreationStackTraces = enableCreationStackTraces
+        DebugProbesImpl.install()
+        installSignalHandler()
+    }
+
+    internal object DebugProbesTransformer : ClassFileTransformer {
+        override fun transform(
+            loader: ClassLoader,
+            className: String,
+            classBeingRedefined: Class<*>?,
+            protectionDomain: ProtectionDomain,
+            classfileBuffer: ByteArray?
+        ): ByteArray? {
+            val name = className.replace("/", ".")
+            if (name != "kotlin.coroutines.jvm.internal.DebugProbesKt") {
```

Comment:
Performance nit: You can don't need to call `replace` and the whole `val name` here. You can directly compare `className` to `"kotlin/coroutines/jvm/internal/DebugProbesKt"`